src/components/tutorial-card/helpers/with-auth-elements.tsx

```diff
@@ -15,7 +16,12 @@ import { getSpeakableDuration } from './build-aria-label'
  * for authenticated users.
  */
 export function TutorialCardWithAuthElements(props: TutorialCardPropsWithId) {
-	const { id: tutorialId, collectionId, ...rest } = props
+	const {
+		id: tutorialId,
+		collectionId,
+		renderBookmarkConfirmationDialog,
```

Comment:
`TutorialCardWithAuth` now accepts an optional prop `renderBookmarkConfirmationDialog. This allows `/profile/bookmarks` to use the same component as we do elsewhere to render tutorial cards.

This doesn't feel ideal to me, but feels like it might be slightly preferable to doing the inverse (which felt like it would be recreating the logic of `TutorialCardWithAuthElements` in `render-bookmark-cards`).

Reply:
Thought / alternative approach: this component could accept an optional prop, `BookmarkButtonComponent`:

```tsx
<TutorialCardWithAuthElements
	// ...
	BookmarkButtonComponent={BookmarkButtonWithRemoveDialog}
/>
```

6 one way, half dozen the other 🤷

Reply:
Nice, I like this! 💖 Updated to accept a `BookmarkButtonComponent` prop, rather than `renderBookmarkConfirmationDialog` 👍